### DISCUSSION TITLE
Any good suggestions begins here

### DISCUSSION THREAD

**Discussion by vomikan** [moneymanagerex/moneymanagerex#3277] [Ideas]
@TomAff Regarding https://github.com/moneymanagerex/moneymanagerex/issues/3267#issuecomment-821069202

Let discuss ideas here.

**Comment by TomAff** [comment]
OK. Here are some suggestions.

**Database**
Add View "Alldata" again. As I know in newer Versions it's not included. I use this for nearly every Report. It will be a great help for new users.

**Filters**
- Add "Filter by Usercolor"
- Change "saved filters" from check points to Listbox. I think it is more comfortable (I saw this in another version. I think 1.4)
![2021-04-19 20_18_16-Window](https://user-images.githubusercontent.com/46572503/115284327-d484ee00-a14c-11eb-917e-633d4e11ba8d.jpg)

**Transactions**
Add Notes-Field on splittet Transactions (Just one Line maybe 50 Letters)
![2021-04-19 20_26_04-Window](https://user-images.githubusercontent.com/46572503/115288450-bec5f780-a151-11eb-95d4-6427afbe727e.jpg)

**Assets**
My suggestion is a complete new handling. Hope you understand what I mean

_Base - add an asset_
Create an Asset with the following fields
Date, Name, Asset type,
Change in value (should always be "depreciation" so it is not needed in my suggestions)
Rate: change from % to month

_Get the amount:_
New field "Asset" in Transactions and in Splittransactions.
In the field you can choose your asset via Listbox.
The value of the asset is the total of all transactions on which you choose the asset

Example
![2021-04-19 21_00_33-Window](https://user-images.githubusercontent.com/46572503/115289141-8e328d80-a152-11eb-9675-fe7ab04fb625.jpg)
(depreciation is always linear)

**Custom Fields**
Add new Type "Link"
![2021-04-19 21_00_33-Window](https://user-images.githubusercontent.com/46572503/115290239-9c34de00-a153-11eb-9bbd-e3be870a34de.jpg)
If this field is relatet to UDFC01 and shown in the transaction view you can click on the link and it opens the file or directory.
This is an alternative to the attachments. At the moment I save invoices in separate folders on my hard drive. Additionally they are saved in the MMEX-Attachment-Folder.

**Open Positions**
Sometimes I have to book receivables or liabilities. At the moment I use an extra category and have to check that the balance is Zero.

Example:
I have to pay 123 Euro at the doctor and I know that it will be refunded by my insurance. 
I book a withdrawal on category "Receivables Insurance"
Later I book the deposit on Category "Receivables Insurance"
Balace "Receivables Insurance" should be zero.

This is no problem with some postings but this way is confusing if you have more of this at the same time.

Suggestions 1 to proceed:
Add boolean-field "OpenPosition" in Categories / Subcategories => Every Posting on this account will be a open position.

Suggestion 2
Just check a checkbox in transaktion/splittransaction to mark it as an open item.

Compensation of open Items
Separate Menu Item => "Open Positions"
=> Choose Category (or all) => Mark positions you want to compensate
I think Boolean-field "OpenPosition" also needed in Checkingaccount/Splittrasactions.

Maybe you have a better solution.
----------

That’s all so far.

Some suggestions may not be of interest to others.

**Reply by vomikan** [reply]
@TomAff  thank you for your proposals.
>Custom Fields Add new Type "Link"

It may be realised with #3224

Change "saved filters" from check points to Listbox #3298

**Reply by TomAff** [reply]
Sorry, I had guessed that some one these sugesstions had already be made

**Comment by vomikan** [comment]
@whalley  Everyone is asked to add comments to the split categories. As long as we don't change the database structure, this can be implemented by storing JSON in the transaction description. What do you think?

**Reply by TomAff** [reply]
just for my understanding. new features habe to be implemented without adding fields or tables?

**Comment by whalley** [comment]
@TomAff  New fields and/or tables could be added but makes things more complicated, having to deal with DB upgrades and affects more code. 

@vomikan Whilst not ideal, I have no objection to using json in the transaction description. Would obviously need to consider existing non-Json descriptions.

**Comment by vomikan** [comment]
A thought occurred to me. Let's say I take the tram to work every day. and every day I pay $ 1 for it. I lent a friend $ 1 and he promised to return it the other day. At work, I buy a business lunch on weekdays every day. And so on.Each time I find a similar transaction and copy it. But you can make special transactions without specifying the date, which will always be in the account today. You can pick them up and set a date. I would call them phantom. This will be especially useful with debt. A daily reminder of a pending transaction.

This echoes @TomAff's idea of Open Positions

**Reply by TomAff** [reply]
That is nearly the way I do ist now. If I pay something, that I have to get back, I set a new transaction maybe one week later. That works but it is Not so comfortable

**Reply by TomAff** [reply]
It was just an idea. Not a must-have

**Comment by vomikan** [comment]
@TomAff that is for you
[Open_Positions.zip](https://github.com/moneymanagerex/moneymanagerex/files/6345160/Open_Positions.zip)

**Reply by TomAff** [reply]
As I can see you handle this via the transactionnumber. If I do it this way, always have to know the number or what if I pay two invoices with one transaction and get it back on two different day. 
But the Idea is great. I think I will try to use it this kind of way. Mybe with a custom field.
Thanks a lot.

**Reply by TomAff** [reply]
Tested it. You always have to set somthing manuelly to make it work. If you forget this, the open position doesn't exist.
Checking the balance is the best way until now.

**Reply by vomikan** [reply]
@TomAff here is fixed SQL
[Open_Positions_v2.zip](https://github.com/moneymanagerex/moneymanagerex/files/6346434/Open_Positions_v2.zip)

**Reply by vomikan** [reply]
In my opinion, this is a simple and effective solution. All transactions with the same number, if the total is zero, are not displayed in the report.
[Open_Positions_v3.zip](https://github.com/moneymanagerex/moneymanagerex/files/6346506/Open_Positions_v3.zip)

**Reply by TomAff** [reply]
Yes, you're right. Now I get it.
But what if you have a splitted transaction. One is expence and one is a receivable?
Am I right that I only have to adjust the sql-script?

**Reply by vomikan** [reply]
This will significantly complicate the SQL script, but is implementable.

**Reply by TomAff** [reply]
I solved it this way:
- I created a new custom field "OP" TRUE/FALSE (instead of using the transactionnumber)
- Let this field be shown in viel "alldata" as "X" when TRUE

SQL for Report
with m as (
select ID,  strftime ('%d.%m.%Y', DATE) DATE, SUBCATEGORY, AMOUNT, round(sum(AMOUNT),2) BALANCE,  OP, NOTES from ALLDATA
	where CATEGORY="31 - Forderungen / Verbindlichkeiten"
	group by SUBCATEGORY, OP
)
select * from m
	where BALANCE<>0

**Reply by vomikan** [reply]
I just wanted to show the underestimated power of customized reports for such a task. I hope that I helped to look at this problem from a different angle.

**Reply by TomAff** [reply]
It's not the way as it is handled in the accounting department, but for my privat use it is a solution. I will use it this way!
Thanks!

**Comment by btab33** [comment]
When you open a recurrent operation to enter it as a new future transaction in your account each month, you have to clic on OK with the mouse. I wish I could just use enter key for this for each opened operation. But it would need "OK" to be preset by default. 
What do tou think about that idea?

**Reply by vomikan** [reply]
https://github.com/moneymanagerex/moneymanagerex/commit/9c61113769cc3ddab3d8b58d85972fa9db6cdc87

**Comment by blue-baloon** [comment]
It would be nice if we could make custom fields mandatory (cannot enter the transaction without filling this field) for some types of transactions, like 
'payment method' field mandatory in credit transactions
'service description' field mandatory in debt transactions

**Comment by Staci3** [comment]
how about import of ofx/qfx files?     also import button on toolbar would be nice bonus

i used to use MS money and loved it, MS needs to open source it.  anyways a few years ago one of my banks made some changes to download formats that didnt work with money, ended up going with quicken also bought moneyspire and money dance - 2 of those didnt pan out.  i hate quicken. my quicken license just expired, software you hate to use and doesnt work right always feels really expensive.  money manager ex looks most promising of all the alternatives i been playing with, just need import options for OFX/QFX... please add this functionality,

**Reply by ovari** [reply]
https://github.com/moneymanagerex/moneymanagerex/issues/3710
